### PR TITLE
COO intersection kernel: respect value intersection order

### PR DESCRIPTION
**Pull request by nikitaved**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92976
* __->__ #92242

### CODE REVIEW

**Comment by nikitaved** [comment]
@cpuhrsch , could you please stamp this one?

**Review by cpuhrsch — APPROVED** [review]

**Review by amjames — APPROVED** [review]

**Comment by jeanschmidt** [comment]
This commit is breaking VS17 builds, I am not able to get the specific error. Please reach out to Skotch Vail or Andrew Krieger for more details.

**Comment by nikitaved** [comment]
@jeanschmidt , looks like MSVC having hard times with cpp17 I guess? Otherwise the code change is trivial.

**Comment by jeanschmidt** [comment]
Yeah, by reading the code I can't figure out why this is causing the mentioned build errors. This change is trivial and should pass. I hope that you get the support to isolate the issue from the approvers and the names I mentioned.

There is a taskforce ongoing to check if we need to include VS17 on our CI jobs or we can fully ditch its support.

File: test/test_sparse.py
Comment (pearu) [inline test/test_sparse.py:3637]: Was this failing prior to this PR?

**Review by pearu — APPROVED** [review]
I have a question but LGTM!

File: test/test_sparse.py
Comment (nikitaved) [reply test/test_sparse.py:3637]: No, just a missing test for completeness.

**Comment by nikitaved** [comment]
Closing, it is obsolete.

**Pull request closed without merging**
